### PR TITLE
[Compaction] Cache compaction score for tablet to avoid calculating each tablet score in every production

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

Keep tablet `compaction score` in memory to avoid calculating tablet score in every production. After `load`, `clone` and `compaction`，updating the tablet `compaction score`in memory.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -127,8 +127,11 @@ OLAPStatus Compaction::do_compaction_impl(int64_t permits) {
     int64_t now = UnixMillis();
     if (compaction_type() == ReaderType::READER_CUMULATIVE_COMPACTION) {
         _tablet->set_last_cumu_compaction_success_time(now);
+        _tablet->update_cumulative_compaction_score(_tablet->calc_compaction_score(CompactionType::CUMULATIVE_COMPACTION));
```
Comment (acelyc111) [inline be/src/olap/compaction.cpp:130]: I found every call of  `update_cumulative_compaction_score()` 's parameter is `calc_compaction_score()`, what's reason to do this?

Comment (acelyc111) [inline be/src/olap/compaction.cpp:130]: I found every call of  `update_cumulative_compaction_score()` 's parameter is `calc_compaction_score()`, what's reason to do this?

Comment (acelyc111) [inline be/src/olap/compaction.cpp:130]: I found every call of  `update_cumulative_compaction_score()` 's parameter is `calc_compaction_score()`, what's reason to do this?

Comment (weizuo93) [reply be/src/olap/compaction.cpp:130]: > I found every call of `update_cumulative_compaction_score()` 's parameter is `calc_compaction_score()`, what's reason to do this?

Calculating compaction score before updating metrics.  Of course, these two steps can be encapsulated as a function.

**Pull request closed without merging**